jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -329,58 +326,20 @@ public void copyTo(Path destination)
             {
                 // to a directory, preserve the filename
                 Path destPath = destination.resolve(src.getFileName().toString());
-                Files.copy(src, destPath,
-                    StandardCopyOption.COPY_ATTRIBUTES,
-                    StandardCopyOption.REPLACE_EXISTING);
+                IO.copyFile(src, destPath);
             }
             else
             {
                 // to a file, use destination as-is
-                Files.copy(src, destination,
-                    StandardCopyOption.COPY_ATTRIBUTES,
-                    StandardCopyOption.REPLACE_EXISTING);
+                IO.copyFile(src, destination);
             }
             return;
         }
 
         // At this point this PathResource is a directory.
         assert isDirectory();
 
-        BiFunction<Path, Path, Path> resolver = src.getFileSystem().equals(destination.getFileSystem())
-            ? Path::resolve
-            : Resource::resolveDifferentFileSystem;
-
-        try (Stream<Path> entriesStream = Files.walk(src))
-        {
-            for (Iterator<Path> pathIterator = entriesStream.iterator(); pathIterator.hasNext();)
-            {
-                Path path = pathIterator.next();
-                if (src.equals(path))
-                    continue;
-
-                Path relative = src.relativize(path);
-                Path destPath = resolver.apply(destination, relative);
-
-                if (LOG.isDebugEnabled())
-                    LOG.debug("CopyTo: {} > {}", path, destPath);
-                if (Files.isDirectory(path))
-                {
-                    ensureDirExists(destPath);
-                }
-                else
-                {
-                    ensureDirExists(destPath.getParent());
-                    Files.copy(path, destPath, StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING);
-                }
-            }
-        }
-    }
-
-    static Path resolveDifferentFileSystem(Path path, Path relative)
-    {
-        for (Path segment : relative)
-            path = path.resolve(segment.toString());
-        return path;
+        IO.copyDir(src, destination);
     }
 
     void ensureDirExists(Path dir) throws IOException
```

Comment:
Can you also move this helper to the `IO` class?

Reply:
Sure.